### PR TITLE
#6224 empty holders text

### LINKED ISSUE #6224: Empty holders tab
Start collecting NFTs and make your mark at the top!

- formatting same as in explore empty state
image

here is one for it anyway: Empty holder tab? Start collecting NFTs and make your mark at the top!

_Originally posted by @exezbcz in https://github.com/kodadot/nft-gallery/issues/5522#issuecomment-1497707041_

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #6224
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=13Qx65nLd6SwdtjrRyuoEtp9CKXhF651xdHBPaXcvhwKm4N1)

## Screenshot 📸

- [x] My fix has changed UI

## Copilot Summary

Screen Shot 2023-06-14 at 7 39 05 PM

### CODE REVIEW

File: components/collection/activity/ownerInsightsTabs/HolderTab.vue [hunk 1]
```diff
@@ -1,60 +1,73 @@
...
-          <div>
-            <div
-              class="is-size-7 has-text-k-blue is-clickable"
-              @click="toggleNFTDetails(holderId)">
-              {{ $t('activity.nftDetails') }}
-              <NeoIcon
-                :icon="
-                  isNFTDetailsOpen[holderId] ? 'chevron-down' : 'chevron-right'
-                " />
+    <template v-if="displayedHolders.length">
```
Comment (daiagi) [inline components/collection/activity/ownerInsightsTabs/HolderTab.vue:3]: you have :

```
<template v-if=...>
...
<div v-else>
```

for consistency sake i would make both either div or template up to you

File: components/collection/activity/ownerInsightsTabs/HolderTab.vue [hunk 2]
```diff
@@ -1,60 +1,73 @@
 <template>
   <div>
```
Comment (daiagi) [inline components/collection/activity/ownerInsightsTabs/HolderTab.vue:2]: there is no need to render
```
<div ref="target" />
```
in the case of no content

**Review by daiagi — CHANGES_REQUESTED** [review]

**Review by vikiival — CHANGES_REQUESTED** [review]
Already suggested by @daiagi

**Comment by prury** [comment]
Highest Sales NFT images not loading on some networks:
(if its not directly related i can open another issue, let me know)
BSX: 
image

SNEK:
image

**Comment by daiagi** [comment]
> if its not directly related i can open another issue, let me know

@prury  not related to this issue. please open another

**Comment by prury** [comment]
Take this collection for example: https://deploy-preview-6229--koda-canary.netlify.app/bsx/collection/3579963603/activity

it has some holders, one of them its: 
image

He bought it at the price of 1KSM, yet, it doesn't show it in the holders tab: 
image

Investigated because i was feeling strange that all KSM values on holders were 0

**Comment by daiagi** [comment]
@prury 
>He bought it at the price of 1KSM, yet, it doesn't show it in the holders tab:

this also, belongs in another issue. 
Thank you

File: components/collection/activity/ownerInsightsTabs/HolderTab.vue [hunk 3]
```diff
@@ -61,15 +61,14 @@
...
-    </template>
+    </div>
     <div
       v-else
       class="is-flex is-justify-content-center is-align-items-center pt-6 px-2">
       <div class="has-text-grey">
         {{ $t('activity.noHolders') }}
       </div>
     </div>
-    <div ref="target" />
```
Comment (daiagi) [inline components/collection/activity/ownerInsightsTabs/HolderTab.vue:59]: deleting this completely will break scrolling
it is still needed in the case where there is content

**Review by daiagi — CHANGES_REQUESTED** [review]

**Comment by prury** [comment]
> @prury
> 
> > He bought it at the price of 1KSM, yet, it doesn't show it in the holders tab:
> 
> this also, belongs in another issue. I've created one:
> 
>     * [activity owner tab shows wrong info #6242](https://github.com/kodadot/nft-gallery/issues/6242)

thank you so much @daiagi !

**Review by daiagi — APPROVED** [review]

**Comment by prachi00** [comment]
can we merge this?

**Comment by exezbcz** [comment]
Small visual thing, can you make it 2 rows, it has small margins on the sides 


like this: 
image

**Comment by prachi00** [comment]
Screen Shot 2023-06-21 at 10 02 57 PM
@exezbcz done

File: components/collection/activity/ownerInsightsTabs/HolderTab.vue (hunk 3 above)
Comment (vikiival) [reply components/collection/activity/ownerInsightsTabs/HolderTab.vue:59]: Hello @prachi00 👀 👋  seems like you have unresolved convo here 👼 
I do not want to break scroll 🥺

**Review by vikiival — CHANGES_REQUESTED** [review]

**Comment by daiagi** [comment]
@vikiival it was fixed already
you have :heavy_check_mark:  from me :)

**Comment by prury** [comment]
Was testing this one again and for Statemine, some Holders still won't show. 
Examples: 
https://deploy-preview-6229--koda-canary.netlify.app/stmn/collection/u-20/activity
https://deploy-preview-6229--koda-canary.netlify.app/stmn/collection/u-1234123/activity

File: components/collection/activity/ownerInsightsTabs/HolderTab.vue (hunk 3 above)
Comment (prachi00) [reply components/collection/activity/ownerInsightsTabs/HolderTab.vue:59]: @vikiival I solved it, forgot to mark though

**Comment by prachi00** [comment]
> Was testing this one again and for Statemine, some Holders still won't show. Examples: https://deploy-preview-6229--koda-canary.netlify.app/stmn/collection/u-20/activity https://deploy-preview-6229--koda-canary.netlify.app/stmn/collection/u-1234123/activity

probably a part of a different issue, this PR is only for adding the empty text

**Comment by daiagi** [comment]
@prury

@prachi00 is right. It os another issue

Also if you look at the events at the collection you mentioned it is a bit strange.
Almost all the events are transfers but there are no mints

This probably breaks some assumption in the code that collectcs the holders

**Comment by prury** [comment]
> @prury
> 
> @prachi00 is right. It os another issue
> 
> Also if you look at the events at the collection you mentioned it is a bit strange. Almost all the events are transfers but there are no mints
> 
> This probably breaks some assumption in the code that collectcs the holders

yup you two are right, confused with another PR that went to prod already, sorry.

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 20 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $20 @ 5.21 USD/DOT ~ 3.839 $DOT
🧗 [13Qx65nLd6SwdtjrRyuoEtp9CKXhF651xdHBPaXcvhwKm4N1](https://polkadot.subscan.io/account/13Qx65nLd6SwdtjrRyuoEtp9CKXhF651xdHBPaXcvhwKm4N1)
🔗 [0xae67fc23eb5601a73f1d48757d042330adaa9b5426c99be2756d3933a07e801a](https://polkadot.subscan.io/extrinsic/0xae67fc23eb5601a73f1d48757d042330adaa9b5426c99be2756d3933a07e801a)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**